src/modules/ejector/ejector.py

```diff
@@ -0,0 +1,198 @@
+    CHURN_LIMIT_QUOTIENT, FAR_FUTURE_EPOCH, MIN_VALIDATOR_WITHDRAWABILITY_DELAY,
+)
+from src.modules.ejector.data_encode import encode_data
+from src.modules.ejector.prediction import RewardsPredictionService
+from src.modules.ejector.typings import ProcessingState, ReportData
+from src.modules.submodules.consensus import ConsensusModule
+from src.modules.submodules.oracle_module import BaseModule
+from src.providers.consensus.typings import Validator
+from src.typings import BlockStamp, EpochNumber
+from src.utils.abi import named_tuple_to_dataclass
+from src.utils.validator_state import is_validator_active
+from src.web3py.extentions.lido_validators import LidoValidator, NodeOperatorIndex
+from src.web3py.typings import Web3
+
+
+logger = logging.getLogger(__name__)
+
+
+class Ejector(BaseModule, ConsensusModule):
+    """
+    1. Get withdrawals amount
+    2. Get exit prediction
+    Loop:
+     a. Get validator
+     b. Remove withdrawal amount
+     c. Increase order
+     d. Check new withdrawals epoches
+     e. If withdrawals ok - exit
+    3. Decode gotten lido validators
+    4. Send hash + send data
+    """
+    CONSENSUS_VERSION = 1
+    CONTRACT_VERSION = 1
+
+    AVG_EXPECTING_TIME_IN_SWEEP_MULTIPLIER = 0.5
+
+    def __init__(self, w3: Web3):
+        self.report_contract = w3.lido_contracts.validators_exit_bus_oracle
+        super().__init__(w3)
+
+        self.prediction_service = RewardsPredictionService(w3)
+
+    def execute_module(self, blockstamp: BlockStamp):
+        self.process_report(blockstamp)
+        report_blockstamp = self.get_blockstamp_for_report(blockstamp)
+        if report_blockstamp:
+            self.process_report(report_blockstamp)
+
+    @lru_cache(maxsize=1)
+    def build_report(self, blockstamp: BlockStamp) -> tuple:
+        validators = self.get_validators_to_eject(blockstamp)
+
+        data, data_format = encode_data(validators)
+
+        return ReportData(
+            self.CONSENSUS_VERSION,
+            blockstamp.ref_slot,
+            len(validators),
+            data_format,
+            data,
+        ).as_tuple()
+
+    def get_validators_to_eject(self, blockstamp: BlockStamp) -> list[tuple[NodeOperatorIndex, LidoValidator]]:
+        chain_config = self._get_chain_config(blockstamp)
+
+        rewards_speed_per_epoch = self.prediction_service.get_rewards_per_epoch(blockstamp, chain_config)
+        epochs_to_sweep = self._get_sweep_delay_in_epochs(blockstamp)
+
+        to_withdraw_amount = self.get_total_unfinalized_withdrawal_requests_amount(blockstamp)
+        total_current_balance = self._get_total_balance(blockstamp)
+
+        validators_to_eject = []
+        validator_to_eject_balance_sum = 0
+        # ToDo replace it with exit order
+        exit_order: list[tuple[NodeOperatorIndex, LidoValidator]] = []
+        for no_index, validator in exit_order:
+            withdrawal_epoch = self._get_withdrawal_epoch_for_latest_validator(blockstamp, len(validators_to_eject) + 1)
```

Comment:
the method has a bit misleading name, could you possibly improve it and add some docs/explanations